src/common/mfu_daos.c

```diff
@@ -1881,3 +1901,2836 @@ int mfu_daos_flist_sync(
+    hdf5->oid_memtype = 0;
+    hdf5->oid_dspace = 0;
+    hdf5->oid_dset = 0;
+    /* DKEY Data */
+    hdf5->dkey_memtype = 0;
+    hdf5->dkey_vtype = 0;
+    hdf5->dkey_dspace = 0;
+    hdf5->dkey_dset = 0;
+    /* AKEY Data */
+    hdf5->akey_memtype = 0;
+    hdf5->akey_vtype = 0;
+    hdf5->akey_dspace = 0;
+    hdf5->akey_dset = 0;
+    /* dims for dsets */
+    hdf5->oid_dims[0] = 0;
+    hdf5->dkey_dims[0] = 0;     
+    hdf5->akey_dims[0] = 0;     
+    /* data for keys */
+    hdf5->oid_data = NULL;
+    hdf5->dkey_data = NULL;
+    hdf5->akey_data = NULL;
+    hdf5->oid = NULL;
+    hdf5->dk = NULL;
+    hdf5->ak = NULL;
+}
+
+static int serialize_recx_single(struct hdf5_args *hdf5, 
+                                 daos_key_t *dkey,
+                                 daos_handle_t *oh,
+                                 daos_iod_t *iod)
+{
+    /* if iod_type is single value just fetch iod size from source
+     * and update in destination object */
+    int         buf_len = (int)(*iod).iod_size;
+    char        buf[buf_len];
+    d_sg_list_t sgl;
+    d_iov_t     iov;
+    int         rc;
+    herr_t      err;
+
+    /* set sgl values */
+    sgl.sg_nr     = 1;
+    sgl.sg_nr_out = 0;
+    sgl.sg_iovs   = &iov;
+    d_iov_set(&iov, buf, buf_len);
+    rc = daos_obj_fetch(*oh, DAOS_TX_NONE, 0, dkey, 1, iod, &sgl,
+                        NULL, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to fetch object");
+        goto out;
+    }
+    /* write single val record to dataset */
+    err = H5Dwrite(hdf5->single_dset, hdf5->single_dtype, H5S_ALL,
+                   hdf5->single_dspace, H5P_DEFAULT, sgl.sg_iovs[0].iov_buf);
+    if (err < 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to write single record dataset");
+        goto out;
+    }
+out:
+    return rc;
+}
+
+static int serialize_recx_array(struct hdf5_args *hdf5,
+                                daos_key_t *dkey,
+                                daos_key_t *akey,
+                                char *rec_name,
+                                uint64_t *ak_index,
+                                daos_handle_t *oh,
+                                daos_iod_t *iod)
+{
+    int                 rc = 0;
+    int                 i = 0;
+    int                 attr_num = 0;
+    int                 buf_len = 0;
+    int                 path_len = 0;
+    int                 encode_buf_len;
+    uint32_t            number = 5;
+    size_t              nalloc;
+    daos_anchor_t       recx_anchor = {0}; 
+    daos_anchor_t       fetch_anchor = {0}; 
+    daos_epoch_range_t  eprs[5];
+    daos_recx_t         recxs[5];
+    daos_size_t         size = 0;
+    char                attr_name[ATTR_NAME_LEN];
+    char                number_str[ATTR_NAME_LEN];
+    char                attr_num_str[ATTR_NAME_LEN];
+    unsigned char       *encode_buf = NULL;
+    d_sg_list_t         sgl;
+    d_iov_t             iov;
+    hid_t               status = 0;
+
```

Comment:
@dsikich We need to make sure these are initialized to 0, so if we `goto out` before opening/creating them, they are valid integers.
```suggestion

    hdf5->rx_dset = 0;
    hdf5->selection_attr = 0;
    hdf5->rx_memspace = 0;

```